s3cmd

```diff
@@ -583,12 +594,79 @@ def cmd_object_del(args):
 
         if not recursive:
             rc = subcmd_object_del_uri(uri_str)
-        else:
+        elif Config().exclude:
             rc = subcmd_batch_del(uri_str = uri_str)
+        else:
+            rc = subcmd_batch_del_iterative(uri_str = uri_str)
         if not rc:
             return rc
     return EX_OK
 
+def subcmd_batch_del_iterative(uri_str = None, bucket = None):
```

Comment:
Maybe you can profit of this PR to have the following functions:
subcmd_object_del_uri
subcmd_batch_del
subcmd_batch_del_iterative
Take an "uri" object in input, instead of an "uri_str" (/bucket). I think that this would simplify the code a little.